arlas-core/src/main/java/io/arlas/server/managers/CollectionReferenceManager.java

```diff
@@ -54,19 +55,20 @@ public ElasticType getType(CollectionReference collectionReference, String field
         if (elasticType == null) {
             String[] props = field.split("\\.");
             CollectionReferenceDescriptionProperty esField = elasticAdmin.describeCollection(collectionReference).properties.get(props[0]);
+            if (esField == null) {
+                return getUnknownType(field, esField, collectionReference.collectionName, throwException);
+            }
             for (int i=1; i<props.length; i++) {
                 esField = esField.properties.get(props[i]);
+                if (esField == null) {
+                    return getUnknownType(field, esField, collectionReference.collectionName, throwException);
+                }
             }
             if (esField != null) {
```

Comment:
According to Intellij, `esField` is never null here

Reply:
if the key `props[i]` doesn't exist in the `esField.properties` map, then the `get` will return null